File: resources/expected/ros2/isolette/lax/src/isolette_cpp_pkg/src/user_code/thermostat_regulate_temperature_manage_heat_source_mhst_src.cpp
#include "isolette_cpp_pkg/user_headers/thermostat_regulate_temperature_manage_heat_source_mhst_src.hpp"
#include "isolette_cpp_pkg/base_headers/enum_converter.hpp"

//===========================================================
// This file will not be overwritten when re-running Codegen
//===========================================================

//=================================================
//  I n i t i a l i z e    E n t r y    P o i n t
//=================================================
void thermostat_regulate_temperature_manage_heat_source_mhst::initialize()
{
    PRINT_INFO("Initialize Entry Point invoked");

    // Initialize the node
    // Initialize the node's incoming data port values here
    isolette_cpp_pkg_interfaces::msg::TempWstatusimpl current_tempWstatus = isolette_cpp_pkg_interfaces::msg::TempWstatusimpl();
    init_current_tempWstatus(current_tempWstatus);

    isolette_cpp_pkg_interfaces::msg::Tempimpl lower_desired_temp = isolette_cpp_pkg_interfaces::msg::Tempimpl();
    init_lower_desired_temp(lower_desired_temp);

    isolette_cpp_pkg_interfaces::msg::Tempimpl upper_desired_temp = isolette_cpp_pkg_interfaces::msg::Tempimpl();
    init_upper_desired_temp(upper_desired_temp);

    isolette_cpp_pkg_interfaces::msg::RegulatorMode regulator_mode = isolette_cpp_pkg_interfaces::msg::RegulatorMode();
    init_regulator_mode(regulator_mode);

}

//=================================================
//  C o m p u t e    E n t r y    P o i n t
//=================================================
void thermostat_regulate_temperature_manage_heat_source_mhst::timeTriggered()
{
    // Handle communication

    // example receiving messages on data ports
    isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr current_tempWstatus = get_current_tempWstatus();
    PRINT_INFO("Received current_tempWstatus: %s", MESSAGE_TO_STRING(current_tempWstatus));
    isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr lower_desired_temp = get_lower_desired_temp();
    PRINT_INFO("Received lower_desired_temp: %s", MESSAGE_TO_STRING(lower_desired_temp));
    isolette_cpp_pkg_interfaces::msg::Tempimpl::SharedPtr upper_desired_temp = get_upper_desired_temp();
    PRINT_INFO("Received upper_desired_temp: %s", MESSAGE_TO_STRING(upper_desired_temp));
    isolette_cpp_pkg_interfaces::msg::RegulatorMode::SharedPtr regulator_mode = get_regulator_mode();
    PRINT_INFO("Received regulator_mode: %s", MESSAGE_TO_STRING(regulator_mode));

    // Example publishing messages
    isolette_cpp_pkg_interfaces::msg::OnOff heat_control = isolette_cpp_pkg_interfaces::msg::OnOff();
    put_heat_control(heat_control);
}

File: resources/expected/ros2/python-isolette/strict/src/isolette_py_pkg/isolette_py_pkg/user_code/thermostat_regulate_temperature_manage_heat_source_mhst_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rosidl_runtime_py.convert import message_to_yaml
from isolette_py_pkg.base_code.thermostat_regulate_temperature_manage_heat_source_mhst_base_src import thermostat_regulate_temperature_manage_heat_source_mhst_base
from isolette_py_pkg_interfaces.msg import TempWstatusimpl
from isolette_py_pkg_interfaces.msg import Tempimpl
from isolette_py_pkg_interfaces.msg import RegulatorMode
from isolette_py_pkg_interfaces.msg import OnOff
from isolette_py_pkg.base_code.enum_converter import *

#===========================================================
# This file will not be overwritten when re-running Codegen
#===========================================================
class thermostat_regulate_temperature_manage_heat_source_mhst(thermostat_regulate_temperature_manage_heat_source_mhst_base):
    def __init__(self):
        super().__init__()
        # invoke initialize entry point
        self.initialize()

        self.get_logger().info("thermostat_regulate_temperature_manage_heat_source_mhst infrastructure set up")

#=================================================
#  I n i t i a l i z e    E n t r y    P o i n t
#=================================================
    def initialize(self):
        self.get_logger().info("Initialize Entry Point invoked")

        # Initialize the node

        # Initialize the node's incoming data port values here

        # Initialize the node's incoming data port values here
        current_tempWstatus = TempWstatusimpl()
        self.init_current_tempWstatus(current_tempWstatus)

        lower_desired_temp = Tempimpl()
        self.init_lower_desired_temp(lower_desired_temp)

        upper_desired_temp = Tempimpl()
        self.init_upper_desired_temp(upper_desired_temp)

        regulator_mode = RegulatorMode()
        self.init_regulator_mode(regulator_mode)


#=================================================
#  C o m p u t e    E n t r y    P o i n t
#=================================================
    def message_to_string(self, msg):
        yaml_str = message_to_yaml(msg)
        return yaml_str

    def timeTriggered(self):
        # Handle communication

        # Example receiving messages on data ports
        current_tempWstatus = self.get_current_tempWstatus()
        self.get_logger().info(f"Received current_tempWstatus: {self.message_to_string(current_tempWstatus)}")

        lower_desired_temp = self.get_lower_desired_temp()
        self.get_logger().info(f"Received lower_desired_temp: {self.message_to_string(lower_desired_temp)}")

        upper_desired_temp = self.get_upper_desired_temp()
        self.get_logger().info(f"Received upper_desired_temp: {self.message_to_string(upper_desired_temp)}")

        regulator_mode = self.get_regulator_mode()
        self.get_logger().info(f"Received regulator_mode: {self.message_to_string(regulator_mode)}")


        # Example publishing messages
        heat_control = OnOff()
        self.put_heat_control(heat_control)


#=================================================
#  Include any additional declarations here
#=================================================
# Additions within these tags will be preserved when re-running Codegen

# Additions within these tags will be preserved when re-running Codegen
